### DISCUSSION TITLE
Standalone dashboard without login.

### DISCUSSION THREAD

**Discussion by obermeier** [apache/streampipes#1832] [Ideas]
If a dashboard was created, it has the possibility to set the permissions to "Public Element"

![grafik](https://github.com/apache/streampipes/assets/1632575/529491b1-db13-41ef-86e3-ee762f6033f2)

With this permission it is possible to access the dashboard without login, e.g. with the standalone link (/standalone/sp:dashboardmodel:...)

But the data are not shown because the measurement API requires an authentication. (/streampipes-backend/api/v4/datalake/measurements/)

One of my questions is: wath should be the meaning of "Public Element". 
     - Accessible for all users of the system, or also users without an account.

I like the idea to have the possibility to provide dashboards without an account.

**Comment by bossenti** [comment]
Hi @obermeier,

thank you for raising this issue.
First, I think the current behavior should be classified as a bug.
Has this worked at any point @dominikriemer @tenthe?
If not, I'm all for removing this option, at least temporarily.

Which brings us back to the main topic. I agree with you, @obermeier, that a truly public dashboard would be a great feature.
That could be interesting for shop floor scenarios as well.
To realize this, it would have to be possible to assign access rights to specific measures in the data lake, right?
Is this already possible?

**Reply by dominikriemer** [reply]
Hi,
that's right - the meaning of `Public Element` is that the resource is available to all authenticated users.
Before we introduced object-based permission management, it was possible to access the dashboard in standalone mode without permissions.
If we want to support anonymous access to the standalone dashboard (similar to the "share as url" feature of Google Docs or Sharepoint), I think we would need to extend the permission system to allow configuration of anonymous access for all dependent resources (for dashboards: the dashboard, the dashboard widgets and the corresponding data lake measure) and then extend the authentication system.

**Reply by dominikriemer** [reply]
> To realize this, it would have to be possible to assign access rights to specific measures in the data lake, right?
> Is this already possible?

@bossenti That's already implemented, all resources (e.g., adapters, pipelines, data lake measures, dashboards) have an associated object with permission settings (see the objects in the `objectPermissions` view of the user database)

**Reply by bossenti** [reply]
That's great 👍
